packages/trader/src/AppV2/Components/TradeParameters/PayoutPerPoint/payout-per-point.tsx

```diff
@@ -1,41 +1,114 @@
 import React from 'react';
 import clsx from 'clsx';
 import { observer } from 'mobx-react';
-import { TextField } from '@deriv-com/quill-ui';
+import { ActionSheet, TextField } from '@deriv-com/quill-ui';
 import { Localize } from '@deriv/translations';
 import { useTraderStore } from 'Stores/useTraderStores';
 import { getCurrencyDisplayCode } from '@deriv/shared';
 import { Skeleton } from '@deriv/components';
+import { isSmallScreen } from 'AppV2/Utils/trade-params-utils';
+import Carousel from 'AppV2/Components/Carousel';
+import CarouselHeader from 'AppV2/Components/Carousel/carousel-header';
+import TradeParamDefinition from 'AppV2/Components/TradeParamDefinition';
+import PayoutPerPointWheel from './payout-per-point-wheel';
 
 type TPayoutPerPointProps = {
     is_minimized?: boolean;
 };
 
 const PayoutPerPoint = observer(({ is_minimized }: TPayoutPerPointProps) => {
-    const { contract_type, currency, proposal_info } = useTraderStore();
-    const contract_key = contract_type.toUpperCase();
-    const { value: payout_per_point } = proposal_info[contract_key]?.obj_contract_basis || {};
+    const [is_open, setIsOpen] = React.useState(false);
+    const {
+        barrier_1,
+        currency,
+        payout_choices,
+        payout_per_point,
+        setPayoutPerPoint,
+        setV2ParamsInitialValues,
+        v2_params_initial_values,
+    } = useTraderStore();
+
+    const is_small_screen = isSmallScreen();
+    const currency_display_code = getCurrencyDisplayCode(currency);
+    const payout_per_point_list = [...payout_choices]
+        .sort((a, b) => Number(a) - Number(b))
+        .map((payout_per_point: string) => ({
+            value: payout_per_point,
+            label: `${payout_per_point} ${currency_display_code}`,
+        }));
+
+    const action_sheet_content = [
```

Comment:
can we extract this out of the component ?

Reply:
The problem is that it contains not only text content, but a component, that is waiting for `payout_per_point`, `setPayoutPerPoint`, `barrier_1` ,`payout_per_point_list`, `setV2ParamsInitialValues`. If I extract, I need to pass all these props, plus will be logical to extract also in `GrowthRate`, `Multiplier`, `RiskManagement`, `Strike`, `TakeProfit` as all of them have such structure.
What do you think?

Reply:
cool